framework/src/media/audio/audio_manager.c

```diff
@@ -488,28 +531,63 @@ int start_audio_stream_in(void *data, uint32_t frames)
 	int retry = AUDIO_STREAM_RETRY_COUNT;
 	struct pcm *pcm;
 
+	// Todo: Need to check g_actual_audio_out_card_id index.
+	audio_card_info_t *cur_card = &g_audio_in_cards[g_actual_audio_in_card_id];
+
 	if ((ret = get_active_audio_device_pcm(&pcm, INPUT)) != AUDIO_MANAGER_SUCCESS) {
 		return ret;
 	}
 
 	if (g_audio_in_cards[g_actual_audio_in_card_id].status == AUDIO_CARD_PAUSE) {
 		ret = ioctl(pcm_get_file_descriptor(pcm), AUDIOIOC_RESUME, NULL);
 		if (ret < 0) {
-			auddbg("Fail to ioctl AUDIOIOC_RESUME, ret = %d\n", ret);
+			printf("Fail to ioctl AUDIOIOC_RESUME, ret = %d\n", ret);
 			return AUDIO_MANAGER_DEVICE_FAIL;
 		}
 
 		g_audio_in_cards[g_actual_audio_in_card_id].status = AUDIO_CARD_RUNNING;
-		audvdbg("Resume the input audio card!!\n");
+		printf("Resume the input audio card!!\n");
 	}
 
 	do {
-		ret = pcm_readi(pcm, data, frames);
-		audvdbg("Read %d frames\n", ret);
+		if(cur_card->resample.necessary) {
+			unsigned int frames_to_read = (unsigned int)((float)frames * cur_card->resample.ratio);
+			if(frames_to_read > (get_input_frame_count() * cur_card->resample.ratio)) {
+				frames_to_read = (get_input_frame_count() * cur_card->resample.ratio);
+			}
+
+			printf("Read frame for resample: %u to %u\n", frames_to_read, frames);
+
+			ret = pcm_readi(pcm, cur_card->resample.readbuffer, frames_to_read);
```

Comment:
here you adder readi already, hence ignore above comment

Reply:
but not frames_to_read, I think frames is correct...
For example,
app wants 16k samplerate / mono / 8bit, and request 16000 frames (1sec duration), but we should resample 32k
and then 16000 frames will be resampled 32000 frames.